From 813384397a1070400e830bd6583c1a8cf4af712a Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Thu, 22 Aug 2019 14:13:53 +0200
Subject: [PATCH 5/5] lib: media: always fetch meta via the preparser

libvlc_media_fetch_local = 0, this mean the preparser is always used,
therefore meta should be only fetched from the preparser.

(cherry picked from commit 890fb755f4b96258425830c6ae4af94af1c76764)
---
 lib/media.c | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)

diff --git a/lib/media.c b/lib/media.c
index 7c839a5fe0..35dd302853 100644
--- a/lib/media.c
+++ b/lib/media.c
@@ -788,20 +788,12 @@ static int media_parse(libvlc_media_t *media, bool b_async,
         input_item_meta_request_option_t parse_scope = META_REQUEST_OPTION_SCOPE_LOCAL;
         int ret;
 
-        /* Ignore libvlc_media_fetch_local flag since local art will be fetched
-         * by libvlc_MetadataRequest */
-        if (parse_flag & libvlc_media_fetch_network)
-        {
-            ret = libvlc_ArtRequest(libvlc, item,
-                                    META_REQUEST_OPTION_SCOPE_NETWORK);
-            if (ret != VLC_SUCCESS)
-                return ret;
-        }
-
         if (parse_flag & libvlc_media_parse_network)
             parse_scope |= META_REQUEST_OPTION_SCOPE_NETWORK;
         if (parse_flag & libvlc_media_fetch_local)
             parse_scope |= META_REQUEST_OPTION_FETCH_LOCAL;
+        if (parse_flag & libvlc_media_fetch_network)
+            parse_scope |= META_REQUEST_OPTION_FETCH_NETWORK;
         if (parse_flag & libvlc_media_do_interact)
             parse_scope |= META_REQUEST_OPTION_DO_INTERACT;
         ret = libvlc_MetadataRequest(libvlc, item, parse_scope, timeout, media);
-- 
2.20.1

